include/vm_map.h

```diff
@@ -47,6 +47,10 @@ void vm_map_protect(vm_map_t *map, vm_addr_t start, vm_addr_t end,
 vm_map_entry_t *vm_map_add_entry(vm_map_t *map, vm_addr_t start, vm_addr_t end,
                                  vm_prot_t prot);
 
+/* Searches the provided memory map for an empty space of at least the given
+   length. It only considers areas that begin at start address, thus addr is
```

Comment:
Statement "at start address" is actually misleading. I think an explanation you give in another comment is much clearer.